diff -durN nano-3.1.orig/lib/cdefs.h nano-3.1/lib/cdefs.h
--- nano-3.1.orig/lib/cdefs.h	2018-09-19 01:34:41.000000000 +0800
+++ nano-3.1/lib/cdefs.h	2018-10-20 17:50:39.171875000 +0800
@@ -292,7 +292,7 @@
    for portability when this file is used in Gnulib.  */
 #ifndef __nonnull
 # if __GNUC_PREREQ (3,3)
-#  define __nonnull(params) __attribute__ ((__nonnull__ params))
+#  define __nonnull(params) __attribute__ ((__nonnull__ (params)))
 # else
 #  define __nonnull(params)
 # endif
