diff --git a/configure.ac b/configure.ac
index 41a4e4ef..beaf343b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -314,7 +314,7 @@ AC_CHECK_FUNCS(fflush fsync fseeko ftello sigaction)
 AC_CHECK_FUNCS(gettimeofday clock_gettime)
 AC_CHECK_FUNCS(setitimer, HAVE_SETITIMER=yes)
 
-AC_CHECK_FUNCS([cygwin_conv_to_full_win32_path])
+AC_CHECK_FUNCS([cygwin_conv_path])
 
 AC_CHECK_FUNCS(setenv putenv, HAVE_SETENV_OR_PUTENV=yes)
 AC_CHECK_FUNCS(unsetenv)
diff --git a/src/session/download.c b/src/session/download.c
index a0f1c4fd..587305da 100644
--- a/src/session/download.c
+++ b/src/session/download.c
@@ -1069,14 +1069,10 @@ subst_file(char *prog, char *file, char *uri)
 				prog--;
 
 			input = 0;
-#if defined(HAVE_CYGWIN_CONV_TO_FULL_WIN32_PATH)
-#ifdef MAX_PATH
-			char new_path[MAX_PATH];
-#else
-			char new_path[1024];
-#endif
+#if defined(HAVE_CYGWIN_CONV_PATH)
+			unsigned char new_path[PATH_MAX];
 
-			cygwin_conv_to_full_win32_path(replace, new_path);
+			cygwin_conv_path(CCP_POSIX_TO_WIN_A|CCP_ABSOLUTE, file, new_path, PATH_MAX+1);
 			add_to_string(&name, new_path);
 #else
 			if (! truncate || strlen(replace) <= tlen)
diff --git a/src/encoding/lzma.c b/src/encoding/lzma.c
index a7a44238..412c6662 100644
--- a/src/encoding/lzma.c
+++ b/src/encoding/lzma.c
@@ -102,7 +102,7 @@ lzma_read(struct stream_encoded *stream, char *buf, int len)
 		}
 	} while (data->flzma_stream.avail_out > 0);
 
-	assert(len - data->flzma_stream.avail_out == data->flzma_stream.next_out - buf);
+	assert(len - data->flzma_stream.avail_out == (char *)data->flzma_stream.next_out - buf);
 	return len - data->flzma_stream.avail_out;
 }
 
