From 05085cc44bac7209d70a03f79b15bcb68688cd1f Mon Sep 17 00:00:00 2001
From: "atom.long" <atom.long@hotmail.com>
Date: Tue, 19 Jun 2018 15:10:55 +0800
Subject: [PATCH 30/30] fix for checking whether stat takes GNU or BSD format

---
 m4/ctng_prog_stat.m4 | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/m4/ctng_prog_stat.m4 b/m4/ctng_prog_stat.m4
index 8a0b26a6..d6b5034f 100644
--- a/m4/ctng_prog_stat.m4
+++ b/m4/ctng_prog_stat.m4
@@ -6,7 +6,7 @@ AC_DEFUN([CTNG_PROG_STAT_FORMAT],
     [AC_CACHE_CHECK([whether stat takes GNU or BSD format],
          [ctng_cv_stat_flavor],
          [touch conftest
-          chmod 642 conftest
+          chmod 644 conftest
           attr_bsd=$(stat -f '%Lp' conftest 2>conftest.stderr.bsd)
           CTNG_MSG_LOG_ENVVAR([attr_bsd], [stat -f output])
           CTNG_MSG_LOG_FILE([conftest.stderr.bsd])
@@ -14,9 +14,9 @@ AC_DEFUN([CTNG_PROG_STAT_FORMAT],
           CTNG_MSG_LOG_ENVVAR([attr_gnu], [stat -c output])
           CTNG_MSG_LOG_FILE([conftest.stderr.gnu])
           rm -f conftest conftest.stderr.*
-          AS_IF([test "$attr_bsd" = "642"],
+          AS_IF([test "$attr_bsd" = "644"],
               [ctng_cv_stat_flavor=BSD],
-              [test "$attr_gnu" = "642"],
+              [test "$attr_gnu" = "644"],
               [ctng_cv_stat_flavor=GNU],
               [ctng_cv_stat_flavor=unknown])])
      AS_IF([test "$ctng_cv_stat_flavor" = "GNU" ], [$1],
-- 
2.17.1

